e2e.bats

```diff
@@ -30,3 +30,58 @@
   [ $(expr "$output" : '.*code.*401') -ne 0 ]
 }
 
+# NET tests
+
+@test "kw.net.lookupHost: accept" {
+  run kwctl run  --request-path test_data/net/pod_creation.json --settings-path test_data/net/settings-lookuphost.yaml annotated-policy.wasm
+
+  # this prints the output when one the checks below fails
+  echo "output = ${output}"
+
+  [ "$status" -eq 0 ]
+  [ $(expr "$output" : '.*"allowed":true.*') -ne 0 ]
+}
+
+@test "kw.net.lookupHost: reject" {
+  run kwctl run  --request-path test_data/net/pod_creation.json --settings-path test_data/net/settings-lookuphost-reject.yaml annotated-policy.wasm
+
+  # this prints the output when one the checks below fails
+  echo "output = ${output}"
+
+  [ "$status" -eq 0 ]
+  [ $(expr "$output" : '.*"allowed":false,"status":{"message":"Wanted 127.0.0.2, got 127.0.0.1 instead","code":401}.*') -ne 0 ]
+}
+
+# OCI tests
+ 
+@test "kw.oci.verifyPubKeysImage: accept a valid signature" {
+  run kwctl run  --request-path test_data/oci/pod_creation_signed.json --settings-path test_data/oci/settings-verify.yaml annotated-policy.wasm
+
+  # this prints the output when one the checks below fails
+  echo "output = ${output}"
+
+  [ "$status" -eq 0 ]
+  [ $(expr "$output" : '.*"allowed":true.*') -ne 0 ]
+}
+
+@test "kw.oci.verifyPubKeysImage: Reject a valid signature because of missing annotation" {
+  run kwctl run  --request-path test_data/oci/pod_creation_signed.json --settings-path test_data/oci/settings-rejection-with-annot.yaml annotated-policy.wasm
+
+  # this prints the output when one the checks below fails
+  echo "output = ${output}"
+
+  [ "$status" -eq 0 ]
+  [ $(expr "$output" : '.*"allowed":false.*') -ne 0 ]
+  [ $(expr "$output" : '.*Image verification failed: missing signatures.*The following constraints were not satisfied.*annotation: nonexistent.*') -ne 0 ]
```

Comment:
While the rejections are working as expected, the return message from kwctl is too verbose. This is the first time we consume these sdk-go context-aware calls.

In this example:
```console
cel-policy $ kwctl run --request-path test_data/oci/pod_creation_signed.json --settings-path test_data/oci/settings-rejection-with-annot.yaml annotated-policy.wasm
2024-05-16T12:57:26.664794Z  WARN kwctl::run: Policy requires access to Kubernetes resources at evaluation time. During this execution the access to Kubernetes resources is denied. This can cause the policy to not behave properly
2024-05-16T12:57:26.664829Z  WARN kwctl::run: Carefully review which types of Kubernetes resources the policy needs via the `inspect` command, then run the policy using the `--allow-context-aware` flag.
libunwind: __unw_add_dynamic_fde: bad fde: FDE is really a CIE
2024-05-16T12:57:28.297695Z  INFO sigstore::cosign::payload::simple_signing: annotations not satisfied because `optional` attribute is None simple_signing=SimpleSigning { critical: Critical { identity: Identity { docker_reference: "ghcr.io/kubewarden/test-verify-image-signatures" }, image: Image { docker_manifest_digest: "sha256:1d9d3da4c60d27b77bb96bba738319c1c4424853fdd10f65982f9f2ca2422a72" }, type_name: "cosign container image signature" }, optional: None } annotations={"annotation": "nonexistent"}
2024-05-16T12:57:28.298019Z ERROR validate{self=PolicyEvaluator { runtime: "wasi" } settings={"validations": Array [Object {"expression": String("kw.oci.verifyPubKeysImage(variables.image, [variables.pubkey], { 'annotation': 'nonexistent' } ).trusted == true"), "messageExpression": String("'Object ' + object.metadata.name + ' is not accepted, verification of image ' + variables.image + ' failed'"), "reason": String("Unauthorized")}], "variables": Array [Object {"expression": String("'ghcr.io/kubewarden/test-verify-image-signatures:signed'"), "name": String("image")}, Object {"expression": String("'-----BEGIN PUBLIC KEY-----' +\n'MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEALYYVLuPXXcKajo5meRcU+c6/iZe' +\n'Dh+o3ZZs1puVW5FyS1cOusnLn209R0IOFL479vm1iFn+auB41JJZUxfc8g==' +\n'-----END PUBLIC KEY-----'\n"), "name": String("pubkey")}]}}: policy_evaluator::runtimes::callback: callback evaluation failed policy_id="cel-policy" binding="kubewarden" operation="v2/verify" error="Image verification failed: missing signatures\nThe following constraints were not satisfied:\nkind: pubKey\nowner: null\nkey: '-----BEGIN PUBLIC KEY-----MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEALYYVLuPXXcKajo5meRcU+c6/iZeDh+o3ZZs1puVW5FyS1cOusnLn209R0IOFL479vm1iFn+auB41JJZUxfc8g==-----END PUBLIC KEY-----'\nannotations:\n  annotation: nonexistent\n"
{"uid":"1299d386-525b-4032-98ae-1949f69f9cfc","allowed":false,"status":{"message":"internal server error: program exited with code Some(1); stderr set to '2024/05/16 12:57:28 failed to evaluate expression: cannot verify image: Callback evaluation failure: Image verification failed: missing signatures\nThe following constraints were not satisfied:\nkind: pubKey\nowner: null\nkey: '-----BEGIN PUBLIC KEY-----MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEALYYVLuPXXcKajo5meRcU+c6/iZeDh+o3ZZs1puVW5FyS1cOusnLn209R0IOFL479vm1iFn+auB41JJZUxfc8g==-----END PUBLIC KEY-----'\nannotations:\n  annotation: nonexistent\n', error: 'error while executing at wasm backtrace:\n    0: 0x73c51b - <unknown>!<wasm function 11049>\n    1: 0x72f7f9 - <unknown>!<wasm function 10210>\n    2: 0x66e723 - <unknown>!<wasm function 6622>\n    3: 0x64ce7b - <unknown>!<wasm function 6168>\n    4: 0x2a767b - <unknown>!<wasm function 681>\n    5: 0x737979 - <unknown>!<wasm function 10651>\n    6: 0x7400a1 - <unknown>!<wasm function 11478>'","code":500},"auditAnnotations":null,"warnings":null}
```

When failing the verification, the sdk-go returns an overly verbose error.